src/indexation/pht.cpp

```diff
@@ -9,14 +9,14 @@ const ValueType IndexEntry::TYPE = ValueType::USER_DATA;
 void Pht::lookupStep(Prefix p, std::shared_ptr<int> lo, std::shared_ptr<int> hi,
         std::shared_ptr<std::vector<std::shared_ptr<Value>>> vals,
         LookupCallback cb, Dht::DoneCallbackSimple done_cb,
-        std::shared_ptr<unsigned> max_common_prefix_len, bool all_values)
+        std::shared_ptr<unsigned> max_common_prefix_len, int start, bool all_values)
 {
     struct node_lookup_result {
         bool done {false};
         bool is_pht {false};
     };
 
-    auto mid = (*lo + *hi)/2;
+    auto mid = /*(start >= 0) ? start :*/ (*lo + *hi)/2;
```

Comment:
This is not, this is in fact the main part of the code that's commented here. oops :x